main.c

```diff
@@ -98,47 +97,48 @@ static const struct wl_pointer_listener pointer_listener = {
 	.axis = noop,
 };
 
-static void create_pointer(struct slurp_state *state,
-		struct wl_pointer *wl_pointer) {
-	struct slurp_pointer *pointer = calloc(1, sizeof(struct slurp_pointer));
-	if (pointer == NULL) {
-		fprintf(stderr, "allocation failed\n");
-		return;
-	}
-	pointer->state = state;
-	pointer->wl_pointer = wl_pointer;
-	wl_list_insert(&state->pointers, &pointer->link);
-
-	wl_pointer_add_listener(wl_pointer, &pointer_listener, pointer);
+static int min(int a, int b) {
+	return (a < b) ? a : b;
 }
 
-static void destroy_pointer(struct slurp_pointer *pointer) {
-	wl_list_remove(&pointer->link);
-	wl_surface_destroy(pointer->cursor_surface);
-	wl_pointer_destroy(pointer->wl_pointer);
-	free(pointer);
+void seat_get_box(struct slurp_seat *seat, struct slurp_box * result) {
+	result->x = min(seat->pressed_x, seat->x);
+	result->y = min(seat->pressed_y, seat->y);
+	result->width = abs(seat->x - seat->pressed_x);
+	result->height = abs(seat->y - seat->pressed_y);
 }
 
-static int min(int a, int b) {
-	return (a < b) ? a : b;
+static void keyboard_handle_key(void *data, struct wl_keyboard *wl_keyboard,
+	uint32_t serial, uint32_t time, uint32_t key, uint32_t key_state) {
```

Comment:
Style: we indent continuation lines for function blocks with two tabs, to tell the different between the arguments and the body.